tokio-util/src/sync/cancellation_token.rs

```diff
@@ -66,6 +67,26 @@ pin_project! {
     }
 }
 
+pin_project! {
+    /// A Future that is resolved once the corresponding [`CancellationToken`]
+    /// is cancelled.
+    ///
+    /// This is counterpart to [`WaitForCancellationFuture`] where it takes
+    /// [`CancellationToken`] by value instead of using a reference.
```

Comment:
```suggestion
    /// This is the counterpart to [`WaitForCancellationFuture`] that takes
    /// [`CancellationToken`] by value instead of using a reference.
```